src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,332 +1,123 @@
-            var ifNode = ifStatement;
-
-            // In the case that the else clause is actually an else if clause, place the if
-            // statement to be moved in a new block in order to make sure that the else
-            // statement matches the right if statement after the edit.
-            var newIfNodeStatement = ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement ?
-                SyntaxFactory.Block(ifNode.Else.Statement) :
-                ifNode.Else.Statement;
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var syntaxFacts = GetSyntaxFactsService();
 
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var ifNode = (IfStatementSyntax)ifStatement;
 
-            var invertedIf = ifNode.WithCondition(Negate(ifNode.Condition, semanticModel, cancellationToken))
-                      .WithStatement(newIfNodeStatement)
-                      .WithElse(ifNode.Else.WithStatement(ifNode.Statement))
-                      .WithAdditionalAnnotations(Formatter.Annotation);
+            // For single line statment, we swap the TrailingTrivia to preserve the single line
+            StatementSyntax newIfNodeStatement = null;
+            ElseClauseSyntax newElseStatement = null;
 
-            var root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
-            var result = root.ReplaceNode(ifNode, invertedIf);
-            return document.WithSyntaxRoot(result);
-        }
-
-        private bool IsComparisonOfZeroAndSomethingNeverLessThanZero(BinaryExpressionSyntax binaryExpression, SemanticModel semanticModel, CancellationToken cancellationToken)
-        {
-            var canSimplify = false;
-
-            if (binaryExpression.Kind() == SyntaxKind.GreaterThanExpression &&
-                binaryExpression.Right.Kind() == SyntaxKind.NumericLiteralExpression)
-            {
-                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
-                    binaryExpression.Left,
-                    (LiteralExpressionSyntax)binaryExpression.Right,
-                    semanticModel,
-                    cancellationToken);
-            }
-            else if (binaryExpression.Kind() == SyntaxKind.LessThanExpression &&
-                     binaryExpression.Left.Kind() == SyntaxKind.NumericLiteralExpression)
-            {
-                canSimplify = CanSimplifyToLengthEqualsZeroExpression(
-                    binaryExpression.Right,
-                    (LiteralExpressionSyntax)binaryExpression.Left,
-                    semanticModel,
-                    cancellationToken);
-            }
-            else if (binaryExpression.Kind() == SyntaxKind.EqualsExpression &&
-                     binaryExpression.Right.Kind() == SyntaxKind.NumericLiteralExpression)
+            var isMultiLine = ifNode.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia);
```

Comment:
what if you have:

```c#
if (foo) {
   x();
   x();
} else {
   y();
   y();
}
```

That code would say this is not multiline.  Is that intentional?

Reply:
You're right.  I changed the variable to `hasNewLineAfterClosingBrace` because it does not catch all types of multiline statements.  I also added the above code to a test and it passes.